backend/Testing/FwHeadless/MediaFileTests.cs

```diff
@@ -0,0 +1,181 @@
+    // 2. If no filename field is provided, then the filename of the uploaded file (i.e. the .Filename property of IFormFile) will be used as the filename field
+    // 3. If a filename is provided and the file ID had a *different* filename, an error will be returned
+    //    NOTE: In my (Robin's) option, that logic should become "the filename is forcefully set to the original filename", e.g. if I uploaded IMG_2010.JPG and
+    //    I want to replace it with IMG_2011.JPG then I should be allowed to do that without jumping through hoops. (The filename on FwHeadless will remain IMG_2010
+    //    so that FieldWorks projects won't get their file links broken).
+
+    [Fact]
+    public async Task UploadFile_WithNoFilenameField_FilenameTakenFromUploadedFile()
+    {
+        var fileId = await Fixture.PostFile(TestRepoZipPath, loginAs: "admin", expectSuccess: true);
+        var metadata = await Fixture.GetFileMetadata(fileId, loginAs: "admin");
+        metadata.Should().NotBeNull();
+        metadata.Filename.Should().Be(TestRepoZipFilename);
+    }
+
+    [Fact]
+    public async Task UploadFile_TwiceWithDifferentFilenames_FilenameTakenFromFirstUpload()
+    {
+        var fileId = await Fixture.PostFile(TestRepoZipPath);
+        var secondPath = TestRepoZipPath + ".bak";
+        if (File.Exists(secondPath)) File.Delete(secondPath);
+        File.Copy(TestRepoZipPath, secondPath);
+        await Fixture.PutFile(secondPath, fileId);
+        var metadata = await Fixture.GetFileMetadata(fileId, loginAs: "admin");
+        metadata.Should().NotBeNull();
+        metadata.Filename.Should().Be(TestRepoZipFilename);
+        var files = await Fixture.ListFiles(Fixture.ProjectId, loginAs: "admin");
+        (files?.Files ?? []).Should().Contain(Path.Join(fileId.ToString(), TestRepoZipFilename));
+        (files?.Files ?? []).Should().NotContain(Path.Join(fileId.ToString(), secondPath));
+    }
+
+    [Fact]
+    public async Task UploadFile_TwiceWithDifferentFilenamesButOverridingFilename_Works()
+    {
+        var fileId = await Fixture.PostFile(TestRepoZipPath);
+        var secondPath = TestRepoZipPath + ".bak";
+        if (File.Exists(secondPath)) File.Delete(secondPath);
+        File.Copy(TestRepoZipPath, secondPath);
+        await Fixture.PutFile(secondPath, fileId, overrideFilename: TestRepoZipFilename, expectSuccess: true);
+        var metadata = await Fixture.GetFileMetadata(fileId, loginAs: "admin");
+        metadata.Should().NotBeNull();
+        metadata.Filename.Should().Be(TestRepoZipFilename);
+    }
+
+    [Fact]
+    public async Task UploadFile_TooLarge_ThrowsError()
+    {
+        var dummyPath = TestRepoZipPath + ".tooLarge";
+        try
+        {
+            if (File.Exists(dummyPath)) File.Delete(dummyPath);
+            Fixture.CreateDummyFile(dummyPath, 1024 * 1024 * 120); // 120 MB
+            await Fixture.PostFile(dummyPath, expectSuccess: false);
```

Comment:
I'd like to assert the status code here too. Maybe return a tuple of `(Guid fileId, Response response)`

Reply:
NOT yet done in commit 7f680de.

Reply:
Done now.